app/models/stage.rb

```diff
@@ -83,8 +83,10 @@ def current_release?(release)
   end
 
   def create_deploy(user, attributes = {})
+    before_command = attributes.delete(:before_command)
     deploys.create(attributes.merge(release: !no_code_deployed)) do |deploy|
-      deploy.build_job(project: project, user: user, command: script, commit: deploy.reference)
+      commands = before_command.to_s.dup << script
+      deploy.build_job(project: project, user: user, command: commands, commit: deploy.reference)
```

Comment:
I'm still coming back to the idea of the Deploy having a hash of ENV vars that we store as a column in the DB.

That way, the hash can have `{ 'HOSTS' => ['app1.pod12...', 'app2.pod12...'] }`.  And if we wait for 5 minutes to start, we can add hosts to that hash.  That way, we also have a record of which hosts that was run, rather than needing to parse the first line of the commands that are run.